Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -1080,6 +1151,9 @@ public func bsonEquals(_ lhs: BSONValue, _ rhs: BSONValue) -> Bool {
     case let (l as [BSONValue], r as [BSONValue]): // TODO: SWIFT-242
         return l.count == r.count && zip(l, r).reduce(true, { prev, next in prev && bsonEquals(next.0, next.1) })
     case (_ as [Any], _ as [Any]): return false
+    case let (l as Symbol, r as Symbol): return l == r
```

Comment:
side thought (perhaps a ticket producing one): can we add an equality operator to the `BsonValue` protocol and depend on that here for deduplication? Something like:
```
public static func == (lhs: Self, rhs: Self) -> Bool
```

Seems like most of these types already have this thing defined, would be cool to use that generically.